docs/adrs/0003-bulk-import-ui.md

```diff
@@ -0,0 +1,98 @@
+
+* `+` Reduce spread of additional, yet similar, cell types with similar code.
+* `+` Reuse existing cell control mechanisms currently in place with minimal modification.
+* `+` Reuse existing wrapping UI elements - cell title, icons, tabs, etc.
+* `-` Work done to the App cell needs to be made backward compatible with existing functionality.
+* `-` Any "breaking" work (i.e. refactor, etc.) will need to be complete before any releases are possible, making intermediate releases possibly challenging.
+* `-` Adding yet another mode for the App cell will increase the maintenance burden on an already complex code base - see especially the FSM.
+* `-` The draft UI is very complex and may turn the existing code into a deep series of nested logic.
+
+### Implement the UI in a new Cell type
+
+This option will create a new nbextension to serve a new cell type as an extension to the code cell.
+
+* `+` This would mean working with a fresh codebase without existing constraints.
+* `+` Intermediate, minimally functional changes, can be rapidly prototyped and released.
+* `-` To stick with the DRY principle, this would mean refactoring parts of the current app cell to make various components importable (message bus responses, job state management, tab menu, etc.) instead of their integration (really we might want to consider that anyway).
+* `-` Much of the structure of the mockup already functions similar to the existing App Cell and might become duplicated work with additional maintenance burden.
+* `-` This would mean updating other components that make use of Narratives - the Static Narrative interface, Search, and the Dashboard, for example, all introspect Narratives and handle cells based on their type.
+
+### Don't use a Cell, use some other interface
+
+The bulk importer UI work has been mocked up as a cell that becomes a part of the Narrative workflow, but that could change. This option would mean using some other interface rather than a cell to implement bulk import work. 
+
+This has been discussed before, especially in the context of how Narratives are intended to work as a repeatable record of analysis and discussion. Generally, importing files to data objects isn't very useful in trying to repeat work, especially if you don't have access to those files. This will also happen when those files expire on the server and are no longer available. In that case, including one or more import cells is misleading - those cells cannot be run again, implying that the analysis that the Narrative performs cannot be completely reproduced.
+
+* `+` Unimpeded by the constraints around the cell architecture as imposed by both the Jupyter Notebook and KBase codebases.
+* `+` Keeping all import jobs consolidated in one place, outside of cells, would reduce Narrative length and Narrative object size.
+* `+` Removing import cells into a separate interface would have a positive impact on load time.
+* `-` Entrypoint to the interface would be something entirely new.
```

Comment:
Because of this point if we're feeling strongly about this option we should definitely discuss with design what the implications are

Reply:
Completely agreed. From the mockups it feels like cell usage was a design decision that was just assumed. But it doesn't have to be. We can ask the design team if they've considered alternatives.

Reply:
This has come up in previous design discussions but I am not sure that eliminating cell usage has been seriously considered (changing the present paradigm? what?!). I would like to see a more comprehensive examination of pros and cons of taking the import cells out of the narrative. As a user, it's annoying to look at other narratives and have to scroll through a load of cells that represent loading local files.

Reply:
As above - we discussed with PMs and decided that we don't want to rework the entire paradigm at this time.